File: bindings/lua/Polycode/RenderFrame.lua
class "RenderFrame"


function RenderFrame:__getvar(name)
	if name == "viewport" then
		local retVal = Polycode.RenderFrame_get_viewport(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Rectangle"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function RenderFrame:__setvar(name,value)
	if name == "viewport" then
		Polycode.RenderFrame_set_viewport(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function RenderFrame:RenderFrame(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.RenderFrame(unpack(arg))
	end
end

function RenderFrame:__delete()
	if self then Polycode.delete_RenderFrame(self.__ptr) end
end

File: bindings/javascript/Polycode/RenderFrame.js
function RenderFrame(viewport) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.RenderFrame(viewport)
	}
	Object.defineProperties(this, {
		'viewport': { enumerable: true, configurable: true, get: RenderFrame.prototype.__get_viewport, set: RenderFrame.prototype.__set_viewport}
	})
}


RenderFrame.prototype.__get_viewport = function() {
	var retVal = new Rectangle("__skip_ptr__")
	retVal.__ptr = 	Polycode.RenderFrame__get_viewport(this.__ptr)
	return retVal
}

RenderFrame.prototype.__set_viewport = function(val) {
	Polycode.RenderFrame__set_viewport(this.__ptr, val.__ptr)
}

Duktape.fin(RenderFrame.prototype, function (x) {
	if (x === RenderFrame.prototype) {
		return;
	}
	Polycode.RenderFrame__delete(x.__ptr)
})
